backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/ui/web/order/BOMExploderCommand.java

```diff
@@ -130,17 +139,59 @@ public BOMExploderCommand(@Nullable final BOMUse bomToUse,
 			final OrderLineCandidate lineCandidate = initialCandidate.toBuilder()
 					.productId(bomLineProductId)
 					.attributes(attributes)
-					.qty(bomLineQty)
+					.qty(Quantitys.create(bomLineQty, bomUomId))
 					.compensationGroupId(compensationGroupId)
-					.uomId(bomUomId)
 					.explodedFromBOMLineId(bomLineId)
 					.build();
 			result.addAll(this.toBuilder()
 					.initialCandidate(lineCandidate)
 					.build()
-					.execute());  // recurse
+					.execute0());  // recurse
 		}
 
 		return result;
 	}
+
+	private List<OrderLineCandidate> reduceCandidates(final List<OrderLineCandidate> result)
+	{
+		final ArrayList<OrderLineCandidate> reduced = new ArrayList<>();
+		for (final OrderLineCandidate candidate : result)
+		{
+			if (candidate.getQty().signum() == 0)
+			{
+				continue;
+			}
+			if (candidate.getQty().signum() < 0)
+			{
+				throw new AdempiereException("Invalid qty: " + candidate.getQty());
+			}
+			if (candidate.getQty().toBigDecimal().compareTo(BigDecimal.ZERO) == 0)
+			{
+				continue;
+			}
+
+			final Optional<OrderLineCandidate> orderLineCandidate = reduced.stream()
+					.filter(cand -> cand.getProductId().equals(candidate.getProductId()))
```

Comment:
to avoid common mistakes i prefer using ProductId.equals(id1, id2) method,
instead of the crap legacy error prone Object.equals(Object) method.

Reply:
Makes sense. Will keep in mind.